junit-platform-launcher/src/main/java/org/junit/platform/launcher/core/DiscoveryIssueCollector.java

```diff
@@ -67,7 +67,7 @@ public void selectorProcessed(UniqueId engineId, DiscoverySelector selector, Sel
 		if (result.getStatus() == FAILED) {
 			this.issues.add(DiscoveryIssue.builder(Severity.ERROR, selector + " resolution failed") //
```

Comment:
The same caution that prompted `toSourceSafely`, should probably put a try-catch around the  implicit `toString` call here that may throw.  This may involve a 3rd party implementation.

And I only just now realized that `DiscoverySelector.toString()` should have been implemented in Cucumber.  😉

Reply:
Well, if a `toString()` method throws an exception, I'm happy to let the entire discovery fail for that engine... :hand_over_mouth: